### PR TITLE
🔐 (auth): Protected routes

### PR DESCRIPTION
**Pull request by Asmaa-204**

### CODE REVIEW

File: app/public/oauth/gitHubDark.png
[the remark this answers is not in the document]
Comment (Asmaa-204) [reply app/public/oauth/gitHubDark.png:1]: i know i'm waiting till the dark state is saved in redux or local storage to render github's logo based on it

File: app/src/components/InputField.tsx
[the remark this answers is not in the document]
Comment (Asmaa-204) [reply app/src/components/InputField.tsx:1]: it's your modal XD, it has to be dark when the theme is dark..I didn't touch your code

File: app/src/features/authentication/reset-password/ResetPasswordModal.tsx
[the remark this answers is not in the document]
Comment (Asmaa-204) [reply app/src/features/authentication/reset-password/ResetPasswordModal.tsx:1]: It's just for protecting the routes

[the remark this answers is not in the document]
Comment (Sarah-Soliman-Incorta) [reply app/src/features/authentication/reset-password/ResetPasswordModal.tsx:1]: what do you mean , if the user isn't authenticated they cannot reset their password? i don't think that's how it should be

File: app/src/components/InputField.tsx
[the remark this answers is not in the document]
Comment (Sarah-Soliman-Incorta) [reply app/src/components/InputField.tsx:1]: okay my bad XD

File: app/src/features/authentication/reset-password/ResetPasswordModal.tsx
[the remark this answers is not in the document]
Comment (Asmaa-204) [reply app/src/features/authentication/reset-password/ResetPasswordModal.tsx:1]: Nope, if the use is authenticated he can't react this url unless he logs out
and if he is already here he can't change the url to navigate to a protected route like the home page

[the remark this answers is not in the document]
Comment (Sarah-Soliman-Incorta) [reply app/src/features/authentication/reset-password/ResetPasswordModal.tsx:1]: i was logged out and i couldn't reach the url

[the remark this answers is not in the document]
Comment (Asmaa-204) [reply app/src/features/authentication/reset-password/ResetPasswordModal.tsx:1]: ok ur right..fixed <3

**Review by sarah-kamall — APPROVED** [review]

**Pull request merged**